src/Repository/IdeaRepository.php

```diff
@@ -131,11 +131,35 @@ public function anonymizeFinalizedIdeas(Adherent $author): void
             ->update()
             ->set('idea.author', 'null')
             ->where('idea.author = :author')
-            ->andWhere('idea.status = :status')
+            ->andWhere('idea.finalizedAt <= :now')
             ->setParameter('author', $author)
-            ->setParameter('status', IdeaStatusEnum::FINALIZED)
+            ->setParameter('now', new \DateTime())
             ->getQuery()
             ->execute()
         ;
     }
+
+    public function addStatusFilter(&$qb, string $alias, string $status)
+    {
+        switch ($status) {
+            case IdeaStatusEnum::UNPUBLISHED:
+                $qb->andWhere(sprintf('%s.enabled = 0', $alias));
+                break;
+            case IdeaStatusEnum::DRAFT:
+                $qb->andWhere(sprintf('%s.publishedAt IS NULL', $alias))
+                    ->setParameter('now', new \DateTime())
+                ;
+                break;
+            case IdeaStatusEnum::PENDING:
+                $qb->andWhere(sprintf('%s.finalizedAt > :now', $alias))
+                    ->setParameter('now', new \DateTime())
+                ;
+                break;
+            case IdeaStatusEnum::FINALIZED:
+                $qb->andWhere(sprintf('%s.finalizedAt <= :now', $alias))
```

Comment:
`finalizedAt IS NOT NULL AND finalizedAt <= :now`

Reply:
It's a SQL query, no need to check if `finalizedAt IS NOT NULL`, `finalizedAt <= :now` is sufficient.